src/imperative/cached_op.cc

```diff
@@ -327,9 +327,10 @@ bool CachedOp::SetBackwardGraph(
   node_range = {num_forward_nodes, idx.num_nodes()};
   entry_range = {num_forward_entries, idx.num_node_entries()};
 
+  bool contain_dynamic_shape = false;
```

Comment:
why do we need this? this variable is not used in this function.

Reply:
This variable is used in `CheckAndInferShape` to handle cases where we have unknown shapes during backward. The CI failed on a test case and adding this variable seems fix it. We also have the same variable existed in `CachedOp::SetForwardGraph`

Reply:
CheckAndInferShape sets that variable if passed. In SetForwardGraph, the variable is used later in the same function.

Reply:
Removed the variable and passed CI. Seems we do not need it anymore in our latest fix.

Reply:
update PR description